apps/heft/src/plugins/ApiExtractorPlugin/ApiExtractorRunner.ts

```diff
@@ -0,0 +1,144 @@
+      terminalProvider,
+      '[api-extractor] '
+    );
+
+    return new Terminal(prefixTerminalProvider);
+  }
+
+  public get filename(): string {
+    return __filename;
+  }
+
+  public initializeTerminal(terminalProvider: ITerminalProvider): void {
+    this._terminal = ApiExtractorRunner.getTerminal(terminalProvider);
+  }
+
+  public async invokeAsync(): Promise<void> {
+    const apiExtractor: typeof TApiExtractor = require(this._configuration.apiExtractorPackagePath);
+    const extractorConfig: TApiExtractor.ExtractorConfig = apiExtractor.ExtractorConfig.loadFileAndPrepare(
+      this._configuration.configFileLocation
+    );
+    const extractorOptions: TApiExtractor.IExtractorInvokeOptions = {
+      localBuild: !this._configuration.production,
+      typescriptCompilerFolder: this._configuration.typescriptPackagePath,
+      messageCallback: (message: TApiExtractor.ExtractorMessage) => {
+        switch (message.logLevel) {
+          case apiExtractor.ExtractorLogLevel.Error: {
+            let logMessage: (IColorableSequence | string)[];
+            if (message.sourceFilePath) {
+              const buildFolderRelativeFilePath: string = path.relative(
+                this._configuration.buildFolder,
+                message.sourceFilePath
+              );
+              logMessage = [
+                Colors.red(
+                  `${buildFolderRelativeFilePath}(${message.sourceFileLine},${message.sourceFileColumn})`
+                ),
+                Colors.white(' - '),
+                Colors.gray(`(${message.category})`),
+                ' ',
+                Colors.red(message.text)
+              ];
+            } else {
+              logMessage = [message.text];
+            }
+
+            this._terminal.writeErrorLine(...logMessage);
+            break;
+          }
+
+          case apiExtractor.ExtractorLogLevel.Warning: {
+            let logMessage: (IColorableSequence | string)[];
+            if (message.sourceFilePath) {
+              const buildFolderRelativeFilePath: string = path.relative(
+                this._configuration.buildFolder,
+                message.sourceFilePath
+              );
+              logMessage = [
+                Colors.yellow(
+                  `${buildFolderRelativeFilePath}(${message.sourceFileLine},${message.sourceFileColumn})`
+                ),
+                Colors.white(' - '),
+                Colors.gray(`(${message.category})`),
+                ' ',
+                Colors.yellow(message.text)
+              ];
+            } else {
+              logMessage = [message.text];
+            }
+
+            this._terminal.writeWarningLine(...logMessage);
+            break;
+          }
+
+          case apiExtractor.ExtractorLogLevel.Verbose: {
+            this._terminal.writeVerboseLine(message.text);
+            break;
+          }
+
+          case apiExtractor.ExtractorLogLevel.None:
```

Comment:
`ExtractorLogLevel.None` should never be printed. Its purpose is to allow your callback to suppress messages.